library/nameops.mli

```diff
@@ -50,16 +50,57 @@ val increment_subscript : Id.t -> Id.t
 
 val forget_subscript    : Id.t -> Id.t
 
+module Name : sig
+
+  include module type of struct include Names.Name end
+
+  val fold_right : (Id.t -> 'a -> 'a) -> Name.t -> 'a -> 'a
+  (** [fold_left f a x] is [f a id] if [x] is [Name id], and [a] otherwise. *)
+
+  val fold_left : ('a -> Id.t -> 'a) -> 'a -> Name.t -> 'a
+  (** [fold_left f x a] is [f id a] if [x] is [Name id], and [a] otherwise. *)
+
+  val iter : (Id.t -> unit) -> Name.t -> unit
+  (** [iter f x] does [f id] if [x] equals [Name id], nothing otherwise. *)
+
+  val map : (Id.t -> Id.t) -> Name.t -> t
+  (** [map f x] is [Anonymous] if [x] is [Anonymous] and [Name (f id)] if [x] is [Name id]. *)
+
+  val fold_map : ('a -> Id.t -> 'a * Id.t) -> 'a -> Name.t -> 'a * Name.t
+  (** [fold_map f x a] is [a',Name id'] if [x] is [Name id] and [f a id] is [(a',id')]. *)
+
+  val out : Name.t -> Id.t
+  (** [out] associates [id] to [Name id]. Raises [Failure "Nameops.out_name"] otherwise. *)
+
+  val max : Name.t -> Name.t -> Name.t
+  (** [max x y] returns [x] if [x] is [Name id] and [y] otherwise. *)
+
+  val cons : Name.t -> Id.t list -> Id.t list
+  (** [cons x l] returns [id::l] if [x] is [Name id] and [l] otherwise. *)
+
+end
+
 val out_name : Name.t -> Id.t
 (** [out_name] associates [id] to [Name id]. Raises [Failure "Nameops.out_name"]
     otherwise. *)
 
 val name_fold : (Id.t -> 'a -> 'a) -> Name.t -> 'a -> 'a
+(** @deprecated Same as [Names.Name.fold_right] *)
```

Comment:
Looks like you could use a deprecated warning here. Following the deprecation PR of PMP, the syntax would be

    [@@ocaml.deprecated "Same as [Names.Name.fold_right]."]